### PR TITLE
r.resamp.interp: implement parallelization with OpenMP

### PR DESCRIPTION
**Pull request by aaronsms**
- Update Makefile
- Add memory and nprocs parameter
- Implement parallelization

Checklists before merging:
- [x] code review
- [x] CI passes
- [x] performance section in documentation
- [x] confirm values in test are from the old version (run new tests with old code)
- [x] run tests without OpenMP
- [x] visual check of results with custom data ("looks good" with non NC SPM dataset)
- [x] check that it works with really large data (16B cells)
- [x] run multi-core benchmark (no degraded performance with many threads)
- [x] run one core benchmark on many resolutions or many cell
- [x] rebase to main
- [x] run with valgrind

### CODE REVIEW

**Review by marisn — APPROVED** [review]
I run module on a small dataset and the results seemed to come out just fine.

File: raster/r.resamp.interp/main.c [hunk 1]
```diff
@@ -107,6 +116,27 @@ int main(int argc, char *argv[])
...
 
+    sscanf(nprocs->answer, "%d", &threads);
+    if (threads < 1)
+    {
+      G_fatal_error(_("<%d> is not valid number of threads."), threads);
+    }
+#if defined(_OPENMP)
+    omp_set_num_threads(threads);
+#else
+    threads = 1;
```
[the remark this answers is not in the document]
Comment (aaronsms) [reply raster/r.resamp.interp/main.c:127]: Hmm that sounds good, I'll do that for the other modules to be parallelized as well.

**Comment by HuidaeCho** [comment]
Please add a test script.

**Comment by petrasovaa** [comment]
This is for 7 billion cells. 
results

It does not scale great, but it's still a nice improvement, it can decrease time roughly by half.

I believe this is ready to merge.

**Pull request merged**